tests/security/tpm2/tpm2_env_setup.pm

```diff
@@ -18,19 +18,27 @@ sub run {
     my $self = shift;
     $self->select_serial_terminal;
 
-    # Add user tss, tss is the default user to start tpm2.0 service
-    my $user = "tss";
-    my $passwd = "susetesting";
-    assert_script_run "useradd -d /home/$user -m $user";
-    assert_script_run "echo $user:$passwd | chpasswd";
-
     # Install the tpm2.0 related packages
     # and then start the TPM2 Access Broker & Resource Manager
-    zypper_call "in expect ibmswtpm2 tpm2.0-abrmd tpm2.0-abrmd-devel openssl tpm2-0-tss tpm2-tss-engine tpm2.0-tools";
+    zypper_call("in expect ibmswtpm2 tpm2.0-abrmd tpm2.0-abrmd-devel openssl tpm2-0-tss tpm2-tss-engine tpm2.0-tools");
+
+    # Add user tss, tss is the default user to start tpm2.0 service
+    # However, a daemon user may be created as well during tss related
+    # packages installation, we can re-use it then with some changes.
+
+    my $tss_user = "tss";
+    my $tss_home = script_output("cat /etc/passwd | grep $tss_user | cut -d : -f6");
+
+    # The home directory of the daemon user 'tss' may not be created if tpm device not exist
+    if (script_run("ls $tss_home") != 0) {
+        record_soft_failure("Home direcotry of user $tss_user doesn't exist, we need create it manually");
```

Comment:
softfails need bugrefs

Reply:
```suggestion
        record_soft_failure("Home directory of user $tss_user doesn't exist, we need create it manually");
```

Reply:
Let me file one

Reply:
There should actually be a test in CI which complains about softfailures without bugrefs. I wonder why that doesn't trigger here.

Reply:
https://bugzilla.suse.com/show_bug.cgi?id=1193305